tests/unit/src/test/scala/tests/BspStatusSuite.scala

```diff
@@ -45,6 +60,63 @@ class BspStatusSuite extends BaseSuite {
     assertEquals(client.status, "some other other text")
   }
 
+  test("bsp-error") {
+    cleanWorkspace()
+    Bill.installWorkspace(workspace.resolve("billWorkspace").toNIO, "Bill")
+    def bloopReports = server.server.reports.bloop.getReports()
+    for {
+      _ <- initialize(
+        Map(
+          "bloopWorkspace" ->
+            """|/metals.json
+               |{ "a": { } }
+               |
+               |/a/src/main/scala/Main.scala
+               |object Main {
+               |  val x: Int = 1
+               |}
+               |""".stripMargin,
+          "billWorkspace" ->
+            """|/src/com/App.scala
+               |object App {
+               |  val x: Int = 1
+               |}
+               |""".stripMargin,
+        ),
+        expectError = false,
+      )
+      _ <- server.didOpen("bloopWorkspace/a/src/main/scala/Main.scala")
+      _ = assertNoDiff(client.pollStatusBar(), s"Bloop ${Icons.default.link}")
+      _ = client.statusParams.clear()
+      _ <- server.didOpen("billWorkspace/src/com/App.scala")
+      _ = assertNoDiff(client.pollStatusBar(), s"Bill ${Icons.default.link}")
+      _ = client.statusParams.clear()
```

Comment:
Why do we need to clear? A real client doesn't do it, so ideally we wouldn't need to do it too. The old status should be cleared when new status is sent.

Or currently we don't handle that scenario?

Reply:
It's just the way we collect status messages in tests. It's a queue not a stack, so I clear to get rid off "history". But I guess it would be better to make it into a stack instead or look at the last element.

Reply:
Ach, it's fine if it's in the tests only. :+1: